python/GafferScene/ScriptProcedural.py

```diff
@@ -192,4 +202,48 @@ def __error( plug, source, error ) :
 			error
 		)
 
+	__performanceMonitor = None
+	@classmethod
+	def __ensurePerformanceMonitor( cls ) :
+
+		if cls.__performanceMonitor is not None :
+			return
+
+		cls.__performanceMonitor = Gaffer.PerformanceMonitor()
+		cls.__performanceMonitor.setActive( True )
+
+	@staticmethod
+	def __printItems( items ) :
+
+		if not len( items ) :
+			return
+
+		width = max( [ len( x[0] ) for x in items ] ) + 4
+		for name, value in items :
+			sys.stderr.write( "  {name:<{width}}{value}\n".format( name = name, width = width, value = value ) )
+
+	@classmethod
+	def __printStatisticsItems( cls, stats, key, n ) :
+
+		stats.sort( key = key, reverse = True )
+		items = [ ( x[0].relativeName( x[0].ancestor( Gaffer.ScriptNode ) ), key( x ) ) for x in stats[:n] ]
+		cls.__printItems( items )
+
+	@classmethod
+	def __printPerformance( cls ) :
+
+		## \todo This code is essentially just cribbed from the
```

Comment:
I originally planned to give the monitor some method like you suggest, but I thought better of it. My thinking is that the monitor is about collection, and presentation should be elsewhere, similar to how we separate implementation and UI elsewhere through Gaffer.

I also read these recently, which advocate for more free functions and less member functions :

http://www.gotw.ca/gotw/084.htm
http://www.drdobbs.com/cpp/how-non-member-functions-improve-encapsu/184401197
http://www.drdobbs.com/self-sufficient-headers/184401705

I'm not sure I buy into it completely, but we're already doing it a fair bit with our Algo.h files, and I expect to push things a bit more that way - in particular the GraphComponent class could do with simplifying down to the basics with the rest moved to GraphComponentAlgo.h.

Anyway, I do expect at some point to have some shared presentation code, but it's not at all clear to me what form it should take yet. One thing I'm considering is having a Stats class that is designed to amalgamate statistics from multiple sources, so that the same presentation code can be shared between all of them - possibly the stats app will just become a thin wrapper around that.

In the meantime, till we have more use cases and more stats sources, and since we're talking about a few lines of totally private code, I think this is the most pragmatic course.